### PR TITLE
fix(adminbar): auto-refresh all modal stats and chart every minute (#224)

### LINKED ISSUE #224: [Bug] Realtime modal stats don't auto-refresh (except Online Users)
## Problem Description

The WP Slimstat realtime modal (accessible from the WordPress admin toolbar)
displays four stat cards — Online Users, Sessions Today, Views Today, and
Referrals Today — alongside a bar chart. Only the "Online Users" counter
refreshes automatically every minute (via its existing polling mechanism).
The remaining three stat cards and the chart remain static until the page is
manually refreshed.

## Impact

Users relying on the realtime modal for live analytics cannot monitor active
sessions, page views, or referral traffic without a manual page reload. This
makes the modal misleading: it presents a "REALTIME" indicator but only
partially delivers live data.

## Steps to Reproduce

1. Log in to WordPress admin
2. Open the WP Slimstat realtime modal from the admin toolbar (top bar icon)
3. Wait approximately 60 seconds
4. Observe: the "Online Users" counter updates automatically
5. Observe: "Sessions Today", "Views Today", "Referrals Today" counters and
   the bar chart do NOT update

## Expected Outcome

All content in the realtime modal — Online Users, Sessions Today, Views
Today, Referrals Today, and the bar chart — should refresh automatically
every minute, consistent with the "REALTIME" label shown in the modal UI.

## Actual Outcome

Only "Online Users" polls for updates every minute. Sessions Today, Views
Today, Referrals Today, and the bar chart remain frozen at their initial
load values until the browser page is manually refreshed.

## Environment

| Field | Value |
|-------|-------|
| WP Slimstat version | 5.4.2 |
| OS | Darwin 24.6.0 arm64 |
| Location | WP Admin Toolbar modal |
| Frequency | Consistent — occurs every time |

## Additional Context

The auto-refresh mechanism already exists for Online Users (1-minute
interval). The fix likely involves extending the same polling interval
to also refresh Sessions Today, Views Today, Referrals Today, and the
chart data on each tick.

---

**Reported via:** `jaan-issue-report` skill
**jaan-to version:** 7.0.0
**Session context used:** false

### PR DESCRIPTION
**Pull request by parhumm**
## Summary

- **New AJAX endpoint** `slimstat_get_adminbar_stats` returns all admin bar data (online, sessions, views, referrals, chart) in a single request with 60s transient caching
- **All 4 stat cards + chart** now auto-refresh every minute (previously only "Online Users" refreshed)
- **Performance**: 6 DB queries combined into 2 using conditional aggregates; chart data reuses `LiveAnalyticsReport::get_users_chart_data()` transient cache — ~1 fresh DB query per minute in steady state
- **#221 consistency preserved**: chart data comes from same `LiveAnalyticsReport` method used by Live Analytics page

Closes #224
Duplicate of #223 (closed)

## Changes

| File | What |
|------|------|
| `admin/index.php` | New `get_adminbar_stats()` AJAX handler, extracted `query_online_count()` + `check_ajax_view_capability()` helpers, added element IDs to stat cards, extended `wp_localize_script` with `is_pro` + `i18n` |
| `admin/assets/js/adminbar-realtime.js` | Full rewrite — updates all stats + chart bars from single AJAX call, exposes `window.slimstatUpdateAdminBar()` |
| `admin/assets/js/admin.js` | `slimstat:minute_pulse` now calls new endpoint, shares update function with adminbar-realtime.js |
| `tests/e2e/adminbar-realtime-refresh.spec.ts` | 6 E2E tests: AJAX response validation, element IDs, i18n, global functions, consistency, no PHP errors |
| `tests/perf/adminbar-refresh-perf.js` | k6 load test: smoke (1 VU) + load (10 VUs), SLA p95 < 500ms |

## Test plan

- [x] 6/6 new E2E tests pass
- [x] k6 perf test file ready
- [x] /simplify code review passed
- [x] No regressions in existing test suite
- [ ] Manual: open admin bar modal, wait 60s, verify all 4 stats + chart update
- [ ] Manual: check Network tab — only 1 AJAX call per minute

## Summary by CodeRabbit

* **New Features**
  * Holistic admin bar realtime refresh: full admin bar stats (online, sessions, views, referrals) update together with animated counts.
  * Dynamic stacked chart for Pro users with peak highlighting and tooltips.
  * Improved localization support for rate/compare labels.

* **Bug Fixes / Improvements**
  * Smarter polling to avoid redundant requests and coordinate updates.
  * Clearer admin bar update error messaging.

* **Tests**
  * Added end-to-end and performance tests covering realtime refresh, chart behavior, localization, and endpoint stability.

### CODE REVIEW

**Comment by parhumm** [comment]
## E2E Test Results — All 6 Passed

```
Running 6 tests using 1 worker

  ✓  1  AJAX endpoint slimstat_get_adminbar_stats returns valid data (4.5s)
  ✓  2  admin bar modal has all required element IDs for auto-refresh (3.9s)
  ✓  3  SlimStatAdminBar localized data includes i18n and is_pro (3.8s)
  ✓  4  slimstatUpdateAdminBar global function is exposed (3.7s)
  ✓  5  AJAX response is consistent with server-rendered stat values (3.7s)
  ✓  6  no PHP errors on admin page with admin bar (785ms)

  6 passed (23.0s)
```

Environment: Local by Flywheel, WordPress 6.8, PHP 8.2, Chromium (Playwright), macOS

---

## Why These Tests Are Trustworthy

Each test validates a **different layer** of the fix. If any one layer breaks, a specific test fails — there's no single point of failure that all 6 share.

### Test 1: AJAX endpoint returns valid data
**What it proves**: The new `slimstat_get_adminbar_stats` PHP handler exists, responds to authenticated POST requests, and returns a correctly structured JSON payload with real DB data.

**Why it's reliable**: It authenticates as a real admin user (via Playwright `storageState`), extracts the WordPress nonce from the live page, then calls `admin-ajax.php` — the same path the browser JS takes. It validates types (`Number.isInteger`), ranges (`≥ 0`), and conditional Pro/Free response shape. A broken SQL query, missing handler registration, or nonce mismatch would fail this test.

### Test 2: All required element IDs exist in DOM
**What it proves**: The 8 HTML element IDs that `adminbar-realtime.js` targets for DOM updates are actually rendered by `add_menu_to_adminbar()` in PHP.

**Why it's reliable**: This is the **contract** between PHP (producer) and JS (consumer). Before this fix, 6 of these IDs didn't exist — the JS had nothing to update. If someone removes an `id=` attribute from the PHP template, this test catches it immediately. Uses `toBeAttached()` (not `toBeVisible()`) because the admin bar modal may be collapsed.

### Test 3: Localized script data has i18n + is_pro
**What it proves**: `wp_localize_script` passes all required config to the frontend — `ajax_url`, `security` (nonce), `is_pro` flag, and all 5 i18n translation keys.

**Why it's reliable**: The JS refresh function uses `SlimStatAdminBar.i18n.was_last_day` etc. to format comparison text. If a key is missing from `wp_localize_script`, the JS would show `undefined` instead of translated text. This test reads the actual `window.SlimStatAdminBar` object from a real WordPress page load — not a mock.

### Test 4: Global update functions are exposed
**What it proves**: `window.slimstatUpdateAdminBar()` and `window.slimstatAnimateElement()` are callable functions on the page.

**Why it's reliable**: `admin.js` calls `window.slimstatUpdateAdminBar(response.data)` on minute pulse. If `adminbar-realtime.js` fails to load or doesn't expose these globals, the admin-side refresh silently breaks with no visible error. This test ensures the cross-script contract holds.

### Test 5: AJAX response matches server-rendered values
**What it proves**: The AJAX endpoint returns data consistent with what PHP rendered into the HTML on page load. Same SQL, same caching layer — no drift between initial render and subsequent refreshes.

**Why it's reliable**: Compares server-rendered DOM text against AJAX response within seconds. Also validates that `formatted` values match the regex `^\d[\d,]*$` (proper number strings, not `NaN` or empty). This guards against the #221 scenario where admin bar and Live Analytics showed different data.

### Test 6: No PHP errors on the admin page
**What it proves**: The admin dashboard loads with HTTP 200 and no `Fatal error`, `PHP Warning`, or `Class not found` in the HTML.

**Why it's reliable**: Adding new PHP code (the AJAX handler, extracted helpers, localized script data) can trigger autoload failures, undefined method calls, or syntax errors. This is a smoke test that catches any PHP-level breakage introduced by our changes. It runs against the real WordPress admin — not a stub.

---

### What these tests do NOT cover (manual verification needed)

| Gap | Why not automated | Manual check |
|-----|-------------------|--------------|
| Visual animation on stat update | Playwright can't reliably assert CSS transitions | Watch modal for 60s, verify number animation |
| Exactly 1 AJAX call per minute | Timing-sensitive, flaky in CI | Check Network tab for `slimstat_get_adminbar_stats` calls |
| Chart bar height changes | Requires waiting for real data change | Verify chart bars update after minute boundary |
| Frontend-only polling (non-admin pages) | Test environment is admin-only auth | Visit frontend as logged-in admin, verify polling |

### k6 performance test

A separate k6 load test (`tests/perf/adminbar-refresh-perf.js`) validates the endpoint handles 10 concurrent admin users with p95 < 500ms. Run with: `k6 run tests/perf/adminbar-refresh-perf.js`

**Comment by parhumm** [comment]
## Issue #221 Verification — Admin Bar Chart Consistency

Ran the `adminbar-chart-consistency.spec.ts` test suite to verify #221 (admin bar chart data diverging from Live Analytics chart) remains fixed after the #224 changes.

```
Running 5 tests using 1 worker

  ✓  1  admin bar chart data matches Live Analytics AJAX response (5.4s)
  ✓  2  admin bar chart renders without PHP errors after fix (1.3s)
  ✓  3  admin bar chart data values are non-negative integers (894ms)
  ✓  4  admin bar chart uses LiveAnalyticsReport when Pro is active (3.9s)
  -  5  free users see decorative chart with valid structure (skipped — Pro is active)

  1 skipped
  4 passed (17.4s)
```

### What was verified:
1. **Data consistency**: Admin bar chart `data-count` values exactly match the Live Analytics AJAX response (`slimstat_get_live_analytics_data`) — both pull from `LiveAnalyticsReport::get_users_chart_data()` with shared 60s transient cache
2. **No PHP errors**: Admin page loads clean (HTTP 200, no Fatal/Warning/Class-not-found)
3. **Valid chart structure**: All 30 bars have non-negative integer counts, minutes-ago ranges 29→0
4. **Pro path active**: Confirmed chart uses real `LiveAnalyticsReport` data, NOT the decorative placeholder array

### Previous issue (now fixed):
The test was previously failing due to a nonce extraction bug in the test itself (regex expected `nonce:` colon syntax but the free-path PHP template uses `var nonce =` assignment syntax). Fixed in commit `b3e1805a`.
